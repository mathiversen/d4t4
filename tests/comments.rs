use data_parser::{parse, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_parse_object_with_comments() -> Result<()> {
    let markup = indoc!(
        r##"{
            # This is a comment
            "values": {
                "r": "10",
                "g": "10",
                "b": "100" # this is also a comment
            },
            "two": "rgba(${values.r}, ${values.g}, ${values.b}, 0.1)",
            "tree": "${values.r}px"
        }"##
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}
