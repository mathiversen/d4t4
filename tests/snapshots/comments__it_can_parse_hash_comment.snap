---
source: tests/comments.rs
expression: x
---
[
  {
    "values": "10"
  }
]
